src/ssr-link.tsx

```diff
@@ -21,7 +21,7 @@ function getRewrites() {
   return (__rewrites = self.__BUILD_MANIFEST.__rewrites);
 }
 
-function useRewriteSource(path: string, locale: string) {
+function useRewriteSource(path: string, locale: string): string {
   const rewrites = getRewrites();
   const lcPath = `/${locale}${path}`;
   const match = rewrites.find(
```

Comment:
As a next step, the discrimination between server & client rendering ought to be more focused, i.e. implementing two different versions of `useRewrites` rather than `IntlLink`.

Reply:
We could eventually check of we have a `window` object and implement either the client or SSR logic?

Reply:
Not enough, I'm afraid -- the serverside code needs to use Node.js modules like `fs` so it can't really be included in the stuff that gets compiled for the client.

Reply:
It's true. But then I don't see how we could refactor IntLink since we won't be able use the same component for both front and back since we won't be able to load `fs` module in the client.
Maybe, instead of using the fs module we could use another module that could be loaded in the client.

Reply:
That's solved by the custom Webpack config, specifically this: https://github.com/Avansai/next-intl-routes/blob/1eb833437dba981ae934155967fb8015b533c7e6/example/next.config.js#L9-L15

That allows for a different implementation of the same function or component to be used by the client and the server. That was the only clean solution I could find, as the file system is only accessible on the server, and `getClientBuildManifest()` will only work on the client.

My "next step" recommendation here is to narrow down the scope of what's different between the two renderers, so that it's just `useRewrites` rather than the whole component.